test/unit/deployment.bats

```diff
@@ -352,3 +352,67 @@ load _helpers
    actual=$(echo "$object" | yq '.[0].value' | tee /dev/stderr)
    [ "${actual}" = 'value1' ]
 }
+
+#--------------------------------------------------------------------
+# extraEnv values
+
+@test "controller/Deployment: extra env string aren't set by default" {
+    cd `chart_dir`
+    local object=$(helm template  \
+      -s templates/deployment.yaml  \
+      . | tee /dev/stderr |  \
+      yq '.spec.template.spec.containers[1].env | select(documentIndex == 1)' |  \
+      tee /dev/stderr)
+
+    local actual=$(echo "$object" | yq '. | length' | tee /dev/stderr)
+    [ "${actual}" = '3' ]
+}
+
+@test "controller/Deployment: extra env string values can be set" {
+    cd `chart_dir`
+    local object=$(helm template  \
+      -s templates/deployment.yaml  \
+      --set 'controller.manager.extraEnv[0].name=HTTP_PROXY'  \
+      --set 'controller.manager.extraEnv[0].value=http://proxy.example.com/'  \
+      . | tee /dev/stderr |  \
+      yq '.spec.template.spec.containers[1].env | select(documentIndex == 1)' |  \
+      tee /dev/stderr)
+
+    local actual=$(echo "$object" | yq '.[3].name' | tee /dev/stderr)
+    [ "${actual}" = 'HTTP_PROXY' ]
+    actual=$(echo "$object" | yq '.[3].value' | tee /dev/stderr)
+    [ "${actual}" = 'http://proxy.example.com/' ]
+}
+
+@test "controller/Deployment: extra env number values can be set" {
+    cd `chart_dir`
+    local object=$(helm template  \
+      -s templates/deployment.yaml  \
+      --set 'controller.manager.extraEnv[0].name=RANDOM_PORT'  \
+      --set 'controller.manager.extraEnv[0].value=42'  \
+      . | tee /dev/stderr |  \
+      yq '.spec.template.spec.containers[1].env | select(documentIndex == 1)' |  \
+      tee /dev/stderr)
+
+    local actual=$(echo "$object" | yq '.[3].name' | tee /dev/stderr)
```

Comment:
We should probably add a length check here to make sure we are not picking up any unexpected values.

Reply:
Could you elaborate on this? I don't get it, I'm afraid. I am already checking default behaviour (no `extraVar` set) in a dedicated check, am I missing something?